cmd/server/cadence/server.go

```diff
@@ -322,6 +323,8 @@ func (s *server) startService() common.Daemon {
 		daemon, err = matching.NewService(&params)
 	case service.Worker:
 		daemon, err = worker.NewService(&params)
+	case service.ShardManager:
```

Comment:
Maybe a crazy idea, but can we start a new cmd for this instead of pushing everything to a single server?

If we start a single process execution - I don't think we need shard manager. Maybe we can have a noop implementation for the in-memory mode.

Reply:
I think it's easier for now to consider this a service like the others. If we want at some point to get completely rid of ringpop we will need the shard manager running even in local mode (at least if people e.g. want more than one history service). 
I think it should be fine to have it as part of the rest of the services, and I would want it to always operate normally so we don't have a strange special case when there is only one history/matching service.

Reply:
I propose to get rid of the existing setup of a single service with multiple services and move towards separated binaries for frontend/history/matching/workers.

Alternatively, we can have a single binary mode - but for a single process there is no point of having shard manager - since you can only start 1 instance per service - so there will be a single history and matching running.
But that is more of a nit/my preference, then a mandatory thing to do.

Reply:
Discussed offline, and we agree that the current initialization setup is not the best, but we will follow the current pattern for now, and maybe do a refactor later.